src/multidimensional_matrix_operators.jl

```diff
@@ -0,0 +1,198 @@
+        new{Dim,T,NodesType,DType,SourceOfCoefficients}(nodes, on_boundary, normals, weights, weights_boundary, Ds, accuracy_order, source)
+    end
+end
+
+Base.ndims(::MultidimensionalMatrixDerivativeOperator{Dim}) where {Dim} = Dim
+Base.getindex(D::MultidimensionalMatrixDerivativeOperator, i::Int) = D.Ds[i]
+Base.eltype(::MultidimensionalMatrixDerivativeOperator{Dim,T}) where {Dim,T} = T
+
+"""
+    integrate_boundary([func,] u, D::MultidimensionalMatrixDerivativeOperator, dim)
+
+Map the function `func` to the coefficients `u` and integrate along the boundary in direction `dim` with respect to
+the surface quadrature rule associated with the `MultidimensionalMatrixDerivativeOperator` `D`.
+"""
+function integrate_boundary(func, u, D::MultidimensionalMatrixDerivativeOperator, dim)
+    return integrate(func, u, weights_boundary_scaled(D, dim))
+end
+
+integrate_boundary(u, D::MultidimensionalMatrixDerivativeOperator, dim) = integrate_boundary(identity, u, D, dim)
+
+weights_boundary(D::MultidimensionalMatrixDerivativeOperator) = get_weight_boundary.(Ref(D), 1:length(grid(D)))
+weights_boundary_scaled(D::MultidimensionalMatrixDerivativeOperator, dim::Int) = get_weight_boundary_scaled.(Ref(D), Ref(dim), 1:length(grid(D)))
+
+"""
+    mass_matrix_boundary(D::MultidimensionalMatrixDerivativeOperator, dim)
+
+Construct the mass matrix at the boundary of a `MultidimensionalMatrixDerivativeOperator` `D` in direction `dim`.
+The boundary mass matrix is constructed to be mimetic, see
+
+- Jan Glaubitz, Simon-Christian Klein, Jan Nordström, Philipp Öffner (2023)
+  Multi-dimensional summation-by-parts operators for general function spaces: Theory and construction
+  Journal of Computational Physics 491, 112370, [DOI: 10.1016/j.jcp.2023.112370](https://doi.org/10.1016/j.jcp.2023.112370).
+"""
+mass_matrix_boundary(D::MultidimensionalMatrixDerivativeOperator, dim::Int) = Diagonal(weights_boundary_scaled(D, dim))
+
+function get_weight_boundary(D::MultidimensionalMatrixDerivativeOperator, i::Int)
+    @unpack weights_boundary, on_boundary = D
+    N, _ = size(D)
+    @boundscheck begin
+        @argcheck 1 <= i <= N
+    end
+    if !on_boundary[i]
+        return zero(eltype(D))
+    end
+    j = sum(view(on_boundary, 1:i))
+    @inbounds ω = weights_boundary[j]
+    return ω
+end
+
+function get_weight_boundary_scaled(D::MultidimensionalMatrixDerivativeOperator, dim::Int, i::Int)
+    @unpack normals, on_boundary = D
+    if !on_boundary[i]
+        return zero(eltype(D))
+    end
+    ω = get_weight_boundary(D, i)
+    j = sum(view(on_boundary, 1:i))
+    ω * normals[j][dim]
+end
+
+function Base.show(io::IO, D::MultidimensionalMatrixDerivativeOperator)
+    if get(io, :compact, false)
+        summary(io, D)
+    else
+        x = grid(D)
+        print(io, ndims(D), "-dimensional matrix-based first-derivative operator {T=", eltype(D), "}")
+        print(io, " on ", length(x), " nodes")
+    end
+end
+
+# TODO: mul! How? Depends on direction
```

Comment:
Good question. I guess another interface is required - just get the operator for a direction and `mul!` that?

Reply:
I see you solved a similar problem for the `FourierDerivative2DOperator`s by adding a method
```julia
function mul!(dest::AbstractArray{T, 2}, D::FourierDerivativeOperator2D, u::AbstractArray{T, 2}, ::Val{:x}) where {T}
```
and similarly for the y-direction. We could do something similar here, but probably not with `::Val{:x}` and `::Val{:y}` because in theory `MultidimensionalMatrixDerivativeOperator` could be used in any dimension. So we could pass an integer as the last argument indicating the direction. On the other hand it's not so painful to write `mul!(dest, D[dim], u)`.

Reply:
If `mul!(dest, D[dim], u)` can be implemented nicely, I would favor this version.

Reply:
So `mul!(dest, D[dim], u)` already works because `D[dim]` is of type `DType`, which should usually implement `mul!`. Thus, I simply removed the TODO note in [33a80ab](https://github.com/ranocha/SummationByPartsOperators.jl/pull/324/commits/33a80ab943d6618e779795868829839fd28f906e). Is this resolved from you point of view?